### ISSUE TITLE
End of pasted text is auto-selected

### ISSUE THREAD

**Issue by zessx** [rubymaniac/vscode-paste-and-indent#4]
Step to reproduce:
1. Copy **at least 2 lines without any indentation**
2. Paste&Indent them on a **indented with `n` spaces**
3. The pasted text is well indented, but has now the `n` last chars selected

Example (from a new file):

    <?php
    // Copy these two lines
    $bar = 2; 
    $baz = 3;

    if (true) {
      $foo = 1;
      // Paste them here
    }

Here, the indentation is 2 spaces, so the chars `3;` will be selected.

This is really annoying, as it's quite specific, and I need to check every copy/paste to ensure I won't erase parts of my code.

**Comment by rubymaniac** [comment]
I have stumbled upon this issue myself. It is not clear (at least to me) under which conditions the selection happens but it has something to do with selections being unstable but I will have to investigate further. I will update this issue with details!

**Comment by supirman** [comment]
Mine have different behaviour. If I copy and paste `n` lines it will select only 2nd line to `n`-th line. So if if I copy one line it did not select anything at all.

**Comment by rubymaniac** [comment]
@supirman yeah this is intended if you have enabled the `selectAfter` option.

**Comment by supirman** [comment]
It is kinda weird, why it is not select the pasted text instead?

**Comment by rubymaniac** [comment]
@supirman because this option is there if you usually find yourself copypasting code that has the first and the last line on different indentation levels like it is the case for offset sensitive languages (Python, YAML etc).

**Comment by varkor** [comment]
I find this occurs even with `selectAfter` disabled.

**Comment by mignz** [comment]
@rubymaniac have you found a solution?

**Comment by rubymaniac** [comment]
No I haven't had the time to look into it. If anyone is willing to look into it would be awesome.

**Comment by gazugafan** [comment]
Heads up: I just released a new Indent on Paste extension that basically does the same thing as this, but takes a different approach that should hopefully circumvent these odd edge case bugs. It's not well tested yet, so it might have some weird bugs of its own, but feel free to try it out!
https://marketplace.visualstudio.com/items?itemName=gazugafan.vscode-indent-on-paste
